react/components/BuyButton/index.js

```diff
@@ -45,44 +42,60 @@ export class BuyButton extends Component {
   }
 
   handleAddToCart = async () => {
```

Comment:
My overall opinion about that is that it is cool, but perharps the handleAddToCart could be partitioned in some other functions... It got bigger